request-manager/graph/grapher.go

```diff
@@ -177,25 +189,32 @@ func getMissingSets(actual map[string]bool, declared []*spec.NodeSet) []string {
 	return missing
 }
 
-// Builds a sequence graph and performs checks on that sequence.
+// buildSeqGraph builds a sequence graph and performs checks on that sequence.
 // It does not analyze any subsequences. As such, it can't e.g. check validity
 // of `sets` declarations in sequence and conditional nodes.
-// Returns the graph, the set of job args its component nodes set, and an error if any should occur.
+// Returns the graph, the set of job args its component nodes set, and an error
+// if any should occur.
 func buildSeqGraph(seqSpec *spec.Sequence, idgen id.Generator) (seqGraph *Graph, sets map[string]bool, err error) {
 	// The graph we'll be filling in
-	seqGraph, err = newGraph(seqSpec.Name, idgen)
+	seqGraph, err = newSeqGraph(seqSpec.Name, idgen)
 	if err != nil {
 		return nil, nil, err
 	}
-	// newGraph doesn't manager `Order` for us; we'll have to maintain it ourselves in this function
+
+	// We have to maintain `Order` ourselves
 	seqGraph.Order = append(seqGraph.Order, seqGraph.Source)
 
-	// Set of job args available, i.e. sequence args + args set by nodes in graph so far
+	// Set of job args available, i.e. sequence args + args set by nodes in
+	// graph so far.
+	// We start out with just the sequence args.
 	jobArgs := getAllSequenceArgs(seqSpec)
```

Comment:
Let's move this var down closer to where/when it's actually used. This is a good practice in general, with one exception being that sometimes we define some widely-used vars at the very top of a func. But otherwise, it helps when reading code that vars appear "just in time".

Reply:
It's used in the same loop as the other vars created between this and the loop itself. Those vars are used only in the loop, though, whereas this and the ones above it are also returned, so this seems like optimal placement to me.

It's like a sandwich, if that makes sense. If not, ignore this line.

Reply:
I've never heard of a sandwich in this sense, but sure. :-) Ok for now then.